notifications: Make `:reset_timer()` more intuitive.

Elv13 opened the pull request:
Previously it would only work if there already was a timeout to avoid some infinite loops. Now it just delay the new timer to the next loop.

Fix #3147

psychon reviewed (commented):
What if someone does `n.timeout = 42` after calling `reset_timeout`? Do they just deserve what they get?

Anyway, :+1: from me.

Elv13 commented:
> What if someone does n.timeout = 42 after calling reset_timeout? Do they just deserve what they get?

Good point. Fixed. I also added a "test" that doesn't really test the right thing, but at least ensure the code gets called.
